### PR TITLE
feat: Add security-bootstrap-redis service

### LINKED ISSUE #332: Add new security-bootstrap-redis service
@andresrinivasan

### PR DESCRIPTION
**Pull request by andresrinivasan**
Signed-off-by: André Srinivasan <andre@redislabs.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/developer-scripts/blob/master/.github/Contributing.md.

## Issue Number:
Resolves #332 

## What is the new behavior?

Add new security-bootstrap-redis service

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

This PR is dependent on actually adding the service to edgex-go. See [edgex-go issue #2503](https://github.com/edgexfoundry/edgex-go/issues/2503)

## Other information

I'm creating this PR for the purposes of review. Please do not merge yet.

### CODE REVIEW

File: releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml
Comment (lenny-intel) [inline releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml:33]: Please down grade your docker-compose to version 1.27.2.

Latest version has a bug that is generating the `depends_on` section incorrectly for the 3.x spec and this breaks the TAF tests.

File: releases/nightly-build/compose-files/source/docker-compose-nexus-add-security.yml
Comment (lenny-intel) [inline releases/nightly-build/compose-files/source/docker-compose-nexus-add-security.yml:22]: This file was renamed in my previous PR. Please remove.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: releases/nightly-build/compose-files/source/docker-compose-nexus-add-security.yml
Comment (andresrinivasan) [reply releases/nightly-build/compose-files/source/docker-compose-nexus-add-security.yml]: Done

File: releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml
Comment (andresrinivasan) [reply releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml]: Done

**Review by bnevis-i — DISMISSED** [review]
Every database-consuming service needs to take a dependency on security-bootstrap-database.  This new dependency needs to be added into add-security.yml.  Otherwise there is no forcing function by any downstream service to ensure that the bootstrap is run.

**Comment by bnevis-i** [comment]
P.S. I've gotten into using `git commit --fixup 572e5274509c0a81320af52d6ca236e946dfca4d -s -S` so that when the time comes I can do a `git rebase -i --autosquash` without then having to go fixup the commit message.

**Comment by lenny-intel** [comment]
> Every database-consuming service needs to take a dependency on security-bootstrap-database. This new dependency needs to be added into add-security.yml. Otherwise there is no forcing function by any downstream service to ensure that the bootstrap is run.

@bnevis-i , good catch!

File: releases/nightly-build/compose-files/docker-compose-nexus-dev.yml
Comment (andresrinivasan) [reply releases/nightly-build/compose-files/docker-compose-nexus-dev.yml]: Apologies. Removed and added this file to .gitignore.

**Comment by andresrinivasan** [comment]
Latest push addresses issues raised in comments.

**Review by lenny-intel — COMMENTED** [review]
Looks good, but can not merge until edgex-go PR merged and image for service is being produced in nexus.

**Comment by andresrinivasan** [comment]
Squashed

**Comment by lenny-intel** [comment]
@andresrinivasan,  My `compose-builder` refactoring PR has been merged. Please rebase yours.

**Comment by andresrinivasan** [comment]
I see the merge conflict and as far can tell I already corrected for the conflicts. Not sure how to resolve this.

File: compose-builder/add-security.yml
Comment (lenny-intel) [inline compose-builder/add-security.yml:250]: Do we need this `database` section at all. Only thin left is the depends_on which doesn't seem to be needed.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by lenny-intel** [comment]
@andresrinivasan , one more rebase needed for my pin versions PR. Last one for me for a bit.... ;-)

File: compose-builder/add-security.yml
Comment (bnevis-i) [reply compose-builder/add-security.yml:250]: Agree with lenny.  redis-bootstrap needs to depend on database and vault-worker.  This can go.

Comment (andresrinivasan) [reply compose-builder/add-security.yml]: Done

**Review by lenny-intel — COMMENTED** [review]
Looks good, but can't merge until docker image is being created in Nexus.

File: compose-builder/add-security.yml
Comment (lenny-intel) [inline compose-builder/add-security.yml:106]: Image name is wrong. Just confirmed `nexus3.edgexfoundry.org:10004/docker-security-bootstrap-redis-go:master` was created.

change `docker-edgex-security-bootstrap-redis-go` to `docker-security-bootstrap-redis-go:master`

Comment (andresrinivasan) [reply compose-builder/add-security.yml:106]: I'll push before 8p PT.

**Comment by lenny-intel** [comment]
@andresrinivasan , you also need to add the `SECRETSTORE_TOKENFILE: /tmp/edgex/secrets/edgex-security-bootstrap-redis/secrets-token.json` env override. I have tested that this with the name changes (see slack) all works.

File: compose-builder/add-security.yml
Comment (andresrinivasan) [reply compose-builder/add-security.yml]: Done

**Comment by andresrinivasan** [comment]
> @andresrinivasan , you also need to add the `SECRETSTORE_TOKENFILE: /tmp/edgex/secrets/edgex-security-bootstrap-redis/secrets-token.json` env override. I have tested that this with the name changes (see slack) all works.

Done

File: compose-builder/add-security.yml
Comment (lenny-intel) [inline compose-builder/add-security.yml:71]: Almost, but missing `security` in the name. Needs to be `docker-security-secrets-setup-go`

Comment (andresrinivasan) [reply compose-builder/add-security.yml]: Done

**Review by lenny-intel — APPROVED** [review]

File: releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml
Comment (lenny-intel) [inline releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml:494]: Still using wrong name. Did you forget to run `make build`?

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Please run `make build` on your latest chnages.

File: compose-builder/add-security.yml
Comment (lenny-intel) [inline compose-builder/add-security.yml:218]: Wrong name. Extra `security`.  Change to `docker-security-proxy-setup-go`

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: compose-builder/add-security.yml
Comment (andresrinivasan) [reply compose-builder/add-security.yml]: Done

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**
